### PR TITLE
chore: convert hash_location_strategy example to a tested spec

### PR DESCRIPTION
**Pull request by Foxandxss**
This is a follow up on #11088

### CODE REVIEW

**Comment by Foxandxss** [comment]
I pushed a change for the format. Now the snippet to show is a bit different due to clang and there is no way I can put the NgModule import first.

File: modules/@angular/examples/common/location/ts/hash_location_strategy/hash_location_strategy_spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class AppModule {
+}
+// #enddocregion hash_location_strategy
+export function main() {
+  describe('hash_location_strategy examples', () => {
+    let location: Location;
+
+    beforeEach(() => {
+      location = TestBed.configureTestingModule({imports: [AppModule], providers: [Location]})
+                     .get(Location);
```
Comment (alxhub) [inline modules/@angular/examples/common/location/ts/hash_location_strategy/hash_location_strategy_spec.ts]: Maybe inject `LocationStrategy` instead, which should get around the platform-server errors.

**Comment by Foxandxss** [comment]
This one is good to go @alxhub

**Comment by mhevery** [comment]
Closing as it is included in https://github.com/angular/angular/pull/11500

**Pull request closed without merging**